pydicom/dataset.py

```diff
@@ -653,6 +674,100 @@ def __getitem__(self, key):
 
         return self._dict.get(tag)
 
+    def _get_private_slot(self, group, private_creator, create=False):
+        """Return the slot for the given tag and private creator.
+
+        If `create` is set an the private creator does not exist,
+        the private creator tag is added.
+        Note: We ignore the unrealistic case that no free slot is
+        available.
+
+        Parameters
+        ----------
+        group : 32 bit int
+            The group of the private tag to be found.
+
+        private_creator : str
+            The private creator string associated with the tag.
+
+        create : bool
+            If `True` and `private_creator` does not exist, a new private
+            creator tag is added at the next free slot.
+
+        Returns
+        -------
+        32 bit int
+            Element base for the given tag (the last 2 digits are always 0)
+
+        Raises
+        ------
+        KeyError
+            If the private creator tag is not found in the given group and
+            the `create` parameter is not set.
+        """
+        for element in range(0x10, 0x100):
+            private_creator_tag = Tag(group, element)
```

Comment:
I'm a bit bothered by the inefficiency of looping one at a time - for code that creates a lot of these it could get somewhat expensive.  But ... I can't immediately think of a good alternate.  We could track all created ones in a dict but then we'd have to maintain that through any data elements removed, etc. - possible, but messy.

Reply:
I didn't care much about efficiency because I thought that these (both getter and setter) will usually be called manually for specific tags, not somewhere inside a loop. Also, in most cases the tag will be found in the first tag.